Source/WebCore/dom/Position.cpp

```diff
@@ -969,24 +979,25 @@ bool Position::nodeIsUserSelectAll(const Node* node)
 {
     if (!node)
         return false;
-    return node->renderer() && (node->renderer()->style().effectiveUserSelect() == UserSelect::All);
+    CheckedPtr renderer = node->renderer();
+    return renderer && renderer->style().effectiveUserSelect() == UserSelect::All;
 }
 
-Node* Position::rootUserSelectAllForNode(Node* node)
+RefPtr<Node> Position::rootUserSelectAllForNode(Node* node)
 {
     if (!node || !nodeIsUserSelectAll(node))
         return nullptr;
-    Node* parent = node->parentNode();
+    RefPtr parent = node->parentNode();
     if (!parent)
         return node;
 
-    Node* candidateRoot = node;
+    RefPtr candidateRoot = node;
     while (parent) {
         if (!parent->renderer()) {
             parent = parent->parentNode();
             continue;
         }
-        if (!nodeIsUserSelectAll(parent))
+        if (!nodeIsUserSelectAll(parent.get()))
             break;
         candidateRoot = parent;
```

Comment:
`candidateRoot = WTFMove(parent);`

or

`candidateRoot = std::exchange(parent, candidateRoot->parentNode());` and omit the next line.

Reply:
I'd go with WTFMove.